engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsNode.java

```diff
@@ -54,6 +54,7 @@
     description = """
       Compares self with other object and returns True iff `self` is exactly the same as
       the other object, including all its transitively accessible properties or fields.
+      Returns Nothing if self and other object are incomparable.
```

Comment:
Tbh I'd not distinguish those. Why can't we just return `False` and let the `less_...` builtin handle incomparable?

Reply:
Good observation. Returning `Nothing` from `EqualsNode` (`Comparable.equals_builtin`) is, indeed, unnecessary and can be left for `Comparable.less_than_builtin`. Refactored.